clients/automation/automation.go

```diff
@@ -195,69 +213,85 @@ func (a Client) List(ctx context.Context, resourceType ResourceType) (api.PagedL
 	var retVal api.PagedListResponse
 
 	// result is the latest API response received
-	var result listResponse
-	result.Count = 1 // ensure starting condition is met, after the first API call this will be actual total count of objects
+	totalCount := 1 // ensure starting condition is met, after the first API call this will be actual total count of objects
 	retrieved := 0
 
 	wfAdminAccess := resourceType == Workflows // only use admin access for workflows
 
-	for retrieved < result.Count {
-		opts := rest.RequestOptions{
-			QueryParams: url.Values{
-				"offset": []string{strconv.Itoa(retrieved)},
-			},
-		}
-		if wfAdminAccess {
-			opts.QueryParams["adminAccess"] = []string{"true"}
+	for retrieved < totalCount {
+		if nextResult, err := a.listPage(ctx, resourceType, wfAdminAccess, retrieved); err != nil {
+			return api.PagedListResponse{}, err
+		} else {
+			if wfAdminAccess != nextResult.WfAdminAccess {
+				wfAdminAccess = nextResult.WfAdminAccess
+				continue
+			}
+
+			totalCount = nextResult.Count
+			retrieved += len(nextResult.Objects)
+			retVal = append(retVal, nextResult.ListResponse)
 		}
+	}
 
-		resp, err := a.restClient.GET(ctx, resources[resourceType].Path, opts)
-		if err != nil {
-			return api.PagedListResponse{}, fmt.Errorf("failed to list automation resources: %w", err)
-		}
+	return retVal, nil
+}
 
-		// if Workflow API rejected the initial request with admin permissions -> retry without
-		if wfAdminAccess && resp.StatusCode == http.StatusForbidden {
-			wfAdminAccess = false
-			resp.Body.Close()
-			continue
-		}
+func (a Client) listPage(ctx context.Context, resourceType ResourceType, wfAdminAccess bool, retrieved int) (listNextResult, error) {
```

Comment:
From the perspective of this function, I think calling it `offset` might be better.

```suggestion
func (a Client) listPage(ctx context.Context, resourceType ResourceType, wfAdminAccess bool, offset int) (listNextResult, error) {
```